### DISCUSSION TITLE
Is there an API to get (value, prefix, unit) ?

### DISCUSSION THREAD

**Discussion by FabriceSalvaire** [hgrecco/pint#1925] [General]
For a particular software (Spice), I need to know for a quantity, the float value, the prefix like pico nano etc., and the unit like meter.
It is because it is not case-sensitive, thus m and M means milli, and mega is Meg.
I could not find the API that permit to properly write a custom formatter.

**Comment by hgrecco** [comment]
You can do something like:

```python
>>> ureg.parse_unit_name("millisecond")
(('milli', 'second', ''),)
```
It returns a list of triplets (prefix, unit, suffix). It is a list because sometimes multiple combinations could match, depending on the definition file. Do you need a custom formatter or a custom parser?
